src/masoniteorm/relationships/HasManyThrough.py

```diff
@@ -57,33 +57,46 @@ def __get__(self, instance, owner):
             if attribute in instance._relationships:
                 return instance._relationships[attribute]
 
-            result = self.apply_query(
+            result = self.apply_related_query(
                 self.distant_builder, self.intermediary_builder, instance
             )
             return result
         else:
             return self
 
-    def apply_query(self, distant_builder, intermediary_builder, owner):
-        """Apply the query and return a dictionary to be hydrated.
-            Used during accessing a relationship on a model
+    def apply_related_query(self, distant_builder, intermediary_builder, owner):
+        """
+        Apply the query to return a Collection of data for the distant models to be hydrated with.
 
-        Arguments:
-            query {oject} -- The relationship object
-            owner {object} -- The current model oject.
+        Method is used when accessing a relationship on a model if its not
+        already eager loaded
 
-        Returns:
-            dict -- A dictionary of data which will be hydrated.
+        Arguments
+            distant_builder (QueryBuilder): QueryBuilder attached to the distant table
+            intermediate_builder (QueryBuilder): QueryBuilder attached to the intermediate (linking) table
+            owner (Any): the model this relationship is starting from
+
+        Returns
+            Collection: Collection of  dicts which will be used for hydrating models.
         """
-        # select * from `countries` inner join `ports` on `ports`.`country_id` = `countries`.`country_id` where `ports`.`port_id` is null and `countries`.`deleted_at` is null and `ports`.`deleted_at` is null
-        result = distant_builder.join(
-            f"{self.intermediary_builder.get_table_name()}",
-            f"{self.intermediary_builder.get_table_name()}.{self.foreign_key}",
-            "=",
-            f"{distant_builder.get_table_name()}.{self.other_owner_key}",
-        ).where(f"{self.intermediary_builder.get_table_name()}.{self.local_owner_key}", getattr(owner, self.other_owner_key)).get()
 
-        return result
+        dist_table = distant_builder.get_table_name()
```

Comment:
lets use full named variables here. distant_table, intermediate_table.

1. shorthand variable names drives me up a wall
2. the common developers attempt to shorthand variable names in an attempt to make code .. ?( more readable?) does the opposite of what it tries to do

Reply:
yep np

Reply:
> lets use full named variables here. distant_table, intermediate_table.
> 
> 1. shorthand variable names drives me up a wall
Yeah I can understand that  ;)

> 2. the common developers attempt to shorthand variable names in an attempt to make code .. ?( more readable?) does the opposite of what it tries to do
True, if its done "too short" it can reduce understanding and readability more than the original more verbose version.

All fixed now